cube/src/main/java/zone/cogni/asquare/cube/convertor/data2shacl/ShaclGenerator.java

```diff
@@ -348,29 +315,42 @@ private Resource calculateNodeKind(boolean hasIri, boolean hasBlank, boolean has
     return null;
   }
 
-  private void setShaclDatatype(Configuration configuration,
-                                RdfStoreService rdfStoreService,
-                                Model shacl,
-                                Resource targetClass,
-                                Resource path,
-                                Resource propertyShape) {
-    List<String> datatypes = selectForTypeAndProperty(rdfStoreService, "select-datatype.sparql.spel",
-                                                      targetClass, path);
+  private void setShaclDatatype(@Nonnull RdfStoreService rdfStore,
+                                @Nonnull Model shacl,
+                                @Nonnull Resource targetClass,
+                                @Nonnull Resource path,
+                                @Nonnull Resource propertyShape) {
+    List<String> datatypes = selectUris(rdfStore, "select-datatype.sparql.spel", getTypeAndPropertyParameters(targetClass, path));
     if (datatypes.size() != 1) {
       String message = getMessage("type '{}' and property '{}' does not have exactly one datatype: {}",
                                   shortenUri(shacl, targetClass), shortenUri(shacl, path), shortenUri(shacl, datatypes));
       log.warn(message);
-
-      if (configuration.isReportPossibleIssues()) {
-        shacl.add(propertyShape, Shacz.warn, message);
-      }
       return;
     }
 
     Resource datatypeValue = ResourceFactory.createResource(datatypes.get(0));
     shacl.add(propertyShape, Shacl.datatype, datatypeValue);
+
+    if (RDF.langString.equals(datatypeValue)) {
+      setLanguageIn(rdfStore, shacl, targetClass, path, propertyShape);
+    }
   }
 
+  private void setLanguageIn(@Nonnull RdfStoreService rdfStore,
+                             @Nonnull Model shacl,
+                             @Nonnull Resource targetClass,
+                             @Nonnull Resource path,
+                             @Nonnull Resource propertyShape) {
+
+    List<Map<String, RDFNode>> rows = getRows(rdfStore,
+                                              "select-languages.sparql.spel",
+                                              getTypeAndPropertyParameters(targetClass, path));
+    List<String> languages = paginatedQuery.convertSingleColumnToList(rows,
+                                                                      input -> input.asLiteral().getString());
```

Comment:
I think you are allowed to just do this (unless you want to be sure it's a Literal, and fail if it's not one.
```suggestion
    List<String> languages = paginatedQuery.convertSingleColumnToList(rows, RDFNode::toString);
```